### PR TITLE
fix: differentiate between persona name and role name in completion

### PR DESCRIPTION
**Pull request by aliel**

### CODE REVIEW

File: src/components/PersonaDialog.vue [hunk 1]
```diff
@@ -103,4 +103,8 @@ watch(
       ipfs_hash: 'QmQMBfgnmuxcQ4kptR1oPE9guYxG13GpASjYVeFQSxNxjE',
     }),
 );
+watch(
+  () => name.value,
+  () => (props.basePersona.label = name.value),
```
Comment (RezaRahemtola) [inline src/components/PersonaDialog.vue]: you should not mutate the prop, and a watch is not even needed here you can pass the label directly in the event emitted.
I'll update this later today

File: src/components/PersonaDropDown.vue [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <template>
   <q-btn-dropdown
     :icon="'img:' + getPersonaAvatarUrl(personasStore.persona.avatar.ipfs_hash)"
-    :label="personasStore.persona.name"
+    :label="personasStore.persona.label"
```
Comment (RezaRahemtola) [inline src/components/PersonaDropDown.vue:4]: why not keep the name here ? Isn't the label for completion only ?

File: src/components/PersonaDropDown.vue [hunk 3]
```diff
@@ -23,7 +23,7 @@
         </q-avatar>
         <q-item-section>
           <q-item-label>
-            {{ persona.name }}
+            {{ persona.label }}
```
Comment (RezaRahemtola) [inline src/components/PersonaDropDown.vue:26]: same

File: src/pages/Chat.vue [hunk 4]
```diff
@@ -32,7 +32,7 @@
             </q-popup-edit>
             <!-- Display the role of the user or the AI -->
             <q-item-label class="text-semibold q-mb-md">
-              {{ message.role }}
+              {{ message.role.replace('assistant', 'Libertai Assistant (default)') }}
```
Comment (RezaRahemtola) [inline src/pages/Chat.vue]: why ? This isn't very generic and clean, we should remove it and fix the underlying problem (if any) in another way

File: src/pages/PersonaManagement.vue [hunk 5]
```diff
@@ -13,7 +13,7 @@
       />
       <persona-dialog
         v-model="createPersona"
-        :base-persona="basePersonaCreate"
+        v-model:base-persona="basePersonaCreate"
```
Comment (RezaRahemtola) [inline src/pages/PersonaManagement.vue]: why?

File: src/pages/PersonaManagement.vue [hunk 6]
```diff
@@ -23,7 +23,7 @@
       />
       <persona-dialog
         v-model="editPersona"
-        :base-persona="personasStore.persona"
+        v-model:base-persona="personasStore.persona"
```
Comment (RezaRahemtola) [inline src/pages/PersonaManagement.vue]: why?

File: src/pages/PersonaManagement.vue [hunk 7]
```diff
@@ -52,7 +52,7 @@
         <q-avatar class="tw-w-24 tw-h-24 tw-mx-auto">
           <img :src="getPersonaAvatarUrl(persona.avatar.ipfs_hash)" alt="avatar" />
         </q-avatar>
-        <q-card-section class="text-bold">{{ persona.name }}</q-card-section>
+        <q-card-section class="text-bold">{{ persona.label }}</q-card-section>
```
Comment (RezaRahemtola) [inline src/pages/PersonaManagement.vue]: why not keep the name?

**Review by RezaRahemtola — CHANGES_REQUESTED** [review]
I think we should reverse the naming, keep name for the global name and label or something else for the internal role completion.

This will be confusing for users and us if we ask them for a "name" and but then it's saved as label and name is something else (in JSON exports for example)

File: src/pages/PersonaManagement.vue (hunk 5 above)
Comment (aliel) [reply src/pages/PersonaManagement.vue]: added automaticaly by npm lint

File: src/pages/PersonaManagement.vue (hunk 6 above)
Comment (aliel) [reply src/pages/PersonaManagement.vue]: added automaticaly by npm lint

File: src/pages/PersonaManagement.vue (hunk 7 above)
Comment (aliel) [reply src/pages/PersonaManagement.vue]: the label field will be changed to "role"

File: src/pages/Chat.vue (hunk 4 above)
Comment (aliel) [reply src/pages/Chat.vue]: should be fixed adding a role field instead of label

File: src/pages/PersonaManagement.vue [hunk 8]
```diff
@@ -13,7 +13,7 @@
       />
       <persona-dialog
         v-model="createPersona"
-        :base-persona="basePersonaCreate"
+        base-persona="basePersonaCreate"
```
Comment (RezaRahemtola) [inline src/pages/PersonaManagement.vue:16]: why?

**Review by RezaRahemtola — APPROVED** [review]

**Pull request merged**